presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -720,10 +731,20 @@ private PlanWithProperties planPartitionedJoin(JoinNode node, List<Symbol> leftS
                 }
                 else {
                     left = withDerivedProperties(
-                            partitionedExchange(idAllocator.getNextId(), REMOTE_STREAMING, left.getNode(), createPartitioning(leftSymbols), Optional.empty()),
+                            partitionedExchange(
```

Comment:
I remember you told me mixed bucket and non-bucketed join is not supported yet since the bucket-to-partition map complicates things. Maybe left a comment/TODO somewhere ? :)

Reply:
Added TODO items